UnityProject/Assets/Scripts/Chat/Chat.cs

```diff
@@ -60,7 +60,17 @@ public static void AddChatMsgToChat(ConnectedPlayer sentByPlayer, string message
 		// This step is skipped when speaking in the OOC channel.
 		(string message, ChatModifier chatModifiers) processedMessage = (string.Empty, ChatModifier.None); // Placeholder values
 		bool isOOC = channels.HasFlag(ChatChannel.OOC);
-		if (!isOOC) processedMessage = ProcessMessage(sentByPlayer, message);
+		if (!isOOC)
+		{
+			processedMessage = ProcessMessage(sentByPlayer, message);
+
+			if (player.mind.occupation.JobType == JobType.MIME && player.mind.IsMiming
```

Comment:
Just an idea - rather than checking based on JobType and IsMiming specifically, could we have a combination of them set an effect on the player instead (muted?) then check for that here? That would then allow us to reuse the functionality for other things - a gag, a "silence" spell etc.

Reply:
being gagged/muted would probably require a different warning message. I'd revisit that when we introduce more reasons for being unable to speak and/or a decent player effects system

Reply:
it would, but the message could be coupled with the effect - that's fine, just from experience having implementation specific details outside of their area will make it harder to maintain in the future. Refactoring when you need too is valid though